etl/steps/data/garden/artificial_intelligence/2024-03-22/papers_with_code_coding.py

```diff
@@ -0,0 +1,56 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+import owid.catalog.processing as pr
+import pandas as pd
+
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    # Load inputs.
+    # Load meadow dataset.
+    ds_meadow = paths.load_dataset("papers_with_code_coding")
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["papers_with_code_coding"].reset_index()
+    # Calculate 'days_since' column.
+    tb["days_since"] = (
+        pd.to_datetime(tb["date"].astype(str), format="%Y-%m-%d") - pd.to_datetime("2019-01-01")
+    ).dt.days
+    tb = tb.drop("date", axis=1)
+
+    grouped_interview = group_sort_filter(tb, "performance_code_any_interview")
+    grouped_competition = group_sort_filter(tb, "performance_code_any_competition")
+
+    tb_combined = pr.merge(grouped_interview, grouped_competition, on=["days_since", "name"], how="outer")
+
+    tb_combined = tb_combined.set_index(["days_since", "name"], verify_integrity=True).sort_index()
+
+    # Save outputs.
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(dest_dir, tables=[tb_combined], default_metadata=ds_meadow.metadata)
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def group_sort_filter(tb, column):
```

Comment:
You could put this into `shared.py` file and import it from all garden datasets to avoid code duplication.